internal/gitinterface/blob.go

```diff
@@ -0,0 +1,38 @@
+package gitinterface
+
+import (
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+func ReadBlob(repo *git.Repository, blobID plumbing.Hash) (int, []byte, error) {
+	blob, err := repo.BlobObject(blobID)
+	if err != nil {
+		return -1, nil, err
+	}
+	contents := make([]byte, blob.Size)
```

Comment:
nit:

```suggestion
	contents := make([]byte, 0, blob.Size)
```

What you have is probably fine assuming that the blob.Size always == the length read, but this is probably safer.